hazelcast/serialization/portable/reader.py

```diff
@@ -400,95 +522,133 @@ def read_char(self, field_name):
+            field_name, FieldType.LONG_ARRAY, super(MorphingPortableReader, self).read_long_array
+        )
 
     def read_float_array(self, field_name):
-        fd = self._class_def.get_field(field_name)
-        if fd is None:
-            return None
-        self.validate_type_compatibility(fd, FieldType.FLOAT_ARRAY)
-        return super(MorphingPortableReader, self).read_float_array(field_name)
+        return self._validate_type_compatibility_and_read(
+            field_name, FieldType.FLOAT_ARRAY, super(MorphingPortableReader, self).read_float_array
+        )
 
     def read_double_array(self, field_name):
-        fd = self._class_def.get_field(field_name)
-        if fd is None:
-            return None
-        self.validate_type_compatibility(fd, FieldType.DOUBLE_ARRAY)
-        return super(MorphingPortableReader, self).read_double_array(field_name)
+        return self._validate_type_compatibility_and_read(
+            field_name,
+            FieldType.DOUBLE_ARRAY,
+            super(MorphingPortableReader, self).read_double_array,
+        )
 
     def read_char_array(self, field_name):
-        fd = self._class_def.get_field(field_name)
-        if fd is None:
-            return None
-        self.validate_type_compatibility(fd, FieldType.CHAR_ARRAY)
-        return super(MorphingPortableReader, self).read_char_array(field_name)
+        return self._validate_type_compatibility_and_read(
+            field_name, FieldType.CHAR_ARRAY, super(MorphingPortableReader, self).read_char_array
+        )
 
     def read_byte_array(self, field_name):
-        fd = self._class_def.get_field(field_name)
-        if fd is None:
-            return None
-        self.validate_type_compatibility(fd, FieldType.BYTE_ARRAY)
-        return super(MorphingPortableReader, self).read_byte_array(field_name)
+        return self._validate_type_compatibility_and_read(
+            field_name, FieldType.BYTE_ARRAY, super(MorphingPortableReader, self).read_byte_array
+        )
 
     def read_boolean_array(self, field_name):
-        fd = self._class_def.get_field(field_name)
-        if fd is None:
-            return None
-        self.validate_type_compatibility(fd, FieldType.BOOLEAN_ARRAY)
-        return super(MorphingPortableReader, self).read_boolean_array(field_name)
+        return self._validate_type_compatibility_and_read(
+            field_name,
+            FieldType.BOOLEAN_ARRAY,
+            super(MorphingPortableReader, self).read_boolean_array,
+        )
 
     def read_portable(self, field_name):
-        fd = self._class_def.get_field(field_name)
-        if fd is None:
-            return None
-        self.validate_type_compatibility(fd, FieldType.PORTABLE)
-        return super(MorphingPortableReader, self).read_portable(field_name)
+        return self._validate_type_compatibility_and_read(
+            field_name, FieldType.PORTABLE, super(MorphingPortableReader, self).read_portable
```

Comment:
You can also replace all usages of `super(MorphingPortableReader, self)` with `super()` in this file to be consistent. These were leftovers from the times when we were Python 2 compatible.

Reply:
Fixed at https://github.com/hazelcast/hazelcast-python-client/pull/569/commits/d641840b5a43ca9d2411c2e1456ed1fce2256fdb. Thanks for the detailed review.